pkg/cmd/roachtest/tests/backup.go

```diff
@@ -634,3 +656,28 @@ func getAWSKMSURI(regionEnvVariable, keyIDEnvVariable string) (string, error) {
 
 	return correctURI, nil
 }
+
+func getGCSKMSURI(keyIDEnvVariable string) (string, error) {
+	q := make(url.Values)
+	expect := map[string]string{
+		"GOOGLE_KMS_CREDENTIALS": gcp.CredentialsParam,
+	}
+	for env, param := range expect {
+		v := os.Getenv(env)
+		if v == "" {
+			return "", errors.Newf("", "%s env var must be set", env)
+		}
+		q.Add(param, v)
+	}
+
+	keyID := os.Getenv(keyIDEnvVariable)
+	if keyID == "" {
+		return "", errors.Newf("", "%s env var must be set", keyIDEnvVariable)
+	}
+
+	// Set AUTH to implicit
+	q.Set(cloudstorage.AuthParam, cloudstorage.AuthParamSpecified)
```

Comment:
ha! The comment above in `getAWSKMSURI` is incorrect. I'm saying use implicit but I'm using specified by sending in the keys as part of the URI. Can you fix that comment please?

As for `getGCSKMSURI` I do think we should use implicit since that is what `backup2TB` etc. use? Then we can get rid of the: 

```
expect := map[string]string{
		"GOOGLE_KMS_CREDENTIALS": gcp.CredentialsParam,
}
```

Reply:
Hmm, so I'm a little confused. Implicit would mean that it looks for a `GOOGLE_APPLICATION_CREDENTIALS` in the env like you were saying above right? But I'm not seeing a `GOOGLE_APPLICATION_CREDENTIALS` in the [nightly builds](https://teamcity.cockroachdb.com/admin/editBuildParams.html?id=buildType:Cockroach_Nightlies_RoachtestNightlyAws).

Also if it is implicit I'd have to give the service account access to the keys right?

Reply:
> But I'm not seeing a GOOGLE_APPLICATION_CREDENTIALS in the nightly builds.

Oh, hmm I think backup2Tb runs on a GCE worker and so it has implicit credentials. I'm referring to https://github.com/adityamaru/cockroach/blob/pts-sql-watcher/pkg/cmd/roachtest/tests/backup.go#L265-L265. This has got me thinking, should we run the GCS version of this test on GCS machines instead? We can just modify the if condition check right at the top to run AWS KMS on AWS, and GCS on GCS?

Reply:
Gotcha, yeah I can try running it on a GCS machine. But I'm still a little unsure how the GCE worker would have permission to use the keys I made. Does it have some sort of admin privilege that gives it access to to everything in GCS or do I have to give it that privilege.?

Reply:
Good question, I'd suggest asking someone in #dev-inf how our nightly google creds are configured. It is possible you might have to make the keys visible to a cockroach-specific project. Another thing you can try is running this roachtest in an actual CI environment, this video [https://cockroachlabs.slack.com/archives/C023S0V4YEB/p1643020977033000] might be useful. If it all gets too much and you think Specified is easier to set up for the time being, I don't want to block you so we can leave this as is, and just fix the comments.